extensions/amp-accordion/1.0/accordion.js

```diff
@@ -397,16 +419,57 @@ export function AccordionContent({
 }) {
   const ref = useRef(null);
   const hasMountedRef = useRef(false);
-  const {contentId, headerId, expanded, animate, setContentId} = useContext(
-    SectionContext
-  );
+  const {
+    contentId,
+    headerId,
+    expanded,
+    animate,
+    setContentId,
+    toggleHandler,
+    experimentDisplayLocking,
+  } = useContext(SectionContext);
   const classes = useStyles();
+  const [supportsContentVisibility, setSupportsContentVisibility] = useState(
+    false
+  );
+  const enableDisplayLocking =
+    supportsContentVisibility && experimentDisplayLocking;
+  const hiddenClass = enableDisplayLocking
+    ? classes.contentHiddenMatchable
+    : classes.contentHidden;
 
   useEffect(() => {
     hasMountedRef.current = true;
     return () => (hasMountedRef.current = false);
   }, []);
 
+  useEffect(() => {
+    const element = ref.current;
+    if (!element) {
+      return;
+    }
+
+    const win = element.ownerDocument.defaultView;
+    if (!win) {
+      return;
+    }
+
+    const newSupportsContentVisibility = win.CSS.supports(
+      'content-visibility',
+      'hidden-matchable'
+    );
+    setSupportsContentVisibility(newSupportsContentVisibility);
+    if (!experimentDisplayLocking || !newSupportsContentVisibility) {
```

Comment:
Perhaps we should test up top `if (experimentDisplayLocking)` and shortcircuit out if not and avoid the `CSS.supports` computation?

Reply:
Updated in the latest revision.